File: frontend/src/api/intelligence.js
import { apiUrl } from "./baseUrl.js";

/**
 * 自然语言 → Cypher → Neo4j → LLM 整理
 * @param {string} question
 * @returns {Promise<{ answer: string, cypher: string | null, records: array | null, meta: object }>}
 */
export async function postIntelligenceQuery(question) {
  const res = await fetch(apiUrl("/api/intelligence/query"), {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ question }),
  });
  let data = {};
  try {
    data = await res.json();
  } catch {
    /* ignore */
  }
  if (!res.ok) {
    const d = data?.detail;
    const msg =
      typeof d === "string"
        ? d
        : Array.isArray(d)
          ? d.map((x) => x.msg ?? JSON.stringify(x)).join("; ")
          : JSON.stringify(data) || res.statusText;
    throw new Error(msg || `请求失败 HTTP ${res.status}`);
  }
  return data;
}

File: backend/app/api/intelligence.py
"""智能问答 API：与 graph 路由分离；未来 replay/simulation 另建路由。"""

from fastapi import APIRouter, HTTPException

from app.intelligence.pipelines.nl_cypher import run_nl_cypher_pipeline
from app.schemas.intelligence import NLQueryRequest, NLQueryResponse

router = APIRouter()


@router.post("/query", response_model=NLQueryResponse)
def nl_query(body: NLQueryRequest) -> NLQueryResponse:
    """
    自然语言 → LLM 生成 Cypher → Neo4j 只读查询 → LLM 整理回答。
    """
    try:
        result = run_nl_cypher_pipeline(body.question)
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e)) from e
    return NLQueryResponse(
        answer=result.answer,
        cypher=result.cypher,
        records=result.records or None,
        meta=result.meta,
    )
